### PR TITLE
Add ability to cancel a Presto query

### PR DESCRIPTION
**Pull request by fbertsch**
@jingw, this passes in our Travis environment and the Dropbox Contributor License Agreement has been filled out.

### CODE REVIEW

File: pyhive/tests/test_presto.py [hunk 1]
```diff
@@ -78,6 +78,17 @@ def test_complex(self, cursor):
         # catch unicode/str
         self.assertEqual(list(map(type, rows[0])), list(map(type, expected[0])))
 
+    @with_cursor
+    def test_cancel(self, cursor):
+        cursor.execute(
+            "SELECT a.a * rand(), b.a*rand()"
```
Comment (jingw) [inline pyhive/tests/test_presto.py]: nitpick: space around *

File: pyhive/presto.py [hunk 2]
```diff
@@ -172,6 +172,16 @@ def execute(self, operation, parameters=None):
...
 
+    def cancel(self):
+        if self._state == self._STATE_NONE:
+            raise ProgrammingError("No query yet")
+        if self._nextUri is None:
+            assert self._state == self._STATE_FINISHED, "Should be finished if nextUri is None"
+            return None
+        response = requests.delete(self._nextUri)
+        self._process_response(response)
+        return response.status_code
```
Comment (fbertsch) [inline pyhive/presto.py]: Sounds good to me. I wasn't completely sure what to return, figured the status code would give some indication of "What you expected to happen, happened".

File: pyhive/presto.py [hunk 3]
```diff
@@ -207,12 +217,18 @@ def _process_response(self, response):
         URI and any data from the response
         """
         # TODO handle HTTP 503
-        if response.status_code != requests.codes.ok:
+        if response.status_code not in (requests.codes.ok, requests.codes.no_content):
             fmt = "Unexpected status code {}\n{}"
             raise OperationalError(fmt.format(response.status_code, response.content))
+
+        if response.status_code == requests.codes.no_content:
```
Comment (fbertsch) [inline pyhive/presto.py]: Will do.

File: pyhive/tests/test_presto.py (hunk 1 above)
Comment (fbertsch) [inline pyhive/tests/test_presto.py]: Should have noticed that myself :)

File: pyhive/common.py [hunk 4]
```diff
@@ -23,6 +23,7 @@ class DBAPICursor(with_metaclass(abc.ABCMeta, object)):
     _STATE_NONE = 0
     _STATE_RUNNING = 1
     _STATE_FINISHED = 2
+    _STATE_CANCELLED = 3
```
Comment (fbertsch) [inline pyhive/common.py]: Sounds good.

**Comment by fbertsch** [comment]
@jingw, the requested changes have been made.

**Comment by jingw** [comment]
Merged, thanks. I made a small tweak to have poll return None instead of throwing an internal assertion error.

**Pull request closed without merging**